src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -997,6 +1028,33 @@ public boolean hasPermission(Permission permission) {
             }
         }
 
+        public static class ExecutableImpl implements Queue.Executable {
```

Comment:
Needs to be an `AsynchronousExecutable` I think, or whatever it is called.

Reply:
Unfortunately I don't understand the idea. Do you mean that ExecutableImpl should extend AsynchronousExecution? Why? Where can I get more info about this question?

Reply:
This is a complicated topic and goes to the root of how Pipeline builds survive Jenkins restarts and specifically how a `sh` step running inside a `node` block will resume. `ExecutorPickle` is used to save the `PlaceholderTask` representing the agent’s executor slot. `PlaceholderExecutable` does some setup when the `node` block is entered, then throws `RunningTask.execution` to represent the fact that it is running without literally consuming a Java thread; this is redone after a restart, hence the check for `cookie == null`.

Reply:
fixed, I guess